src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/PageActionInvoker.cs

```diff
@@ -359,10 +356,10 @@ private Task Next(ref State next, ref Scope scope, ref object state, ref bool is
                 }
             }
 
-            var originalValues = _propertyProvider.LoadAndTrackChanges(_page, _pageContext.TempData);
             if (propertyFilter != null)
             {
-                propertyFilter.OriginalValues = originalValues;
+                propertyFilter.ApplyTempDataChanges(_page, _pageContext.HttpContext);
```

Comment:
@pranavkm - this is not the final design. We still need to spec out filters for pages.